lib/penelope/nlp/intent_classifier.ex

```diff
@@ -90,16 +91,18 @@ defmodule Penelope.NLP.IntentClassifier do
           context :: map,
           x :: String.t()
         ) ::
-          {intent :: String.t(),
+          {intents :: %{(intent :: String.t()) => probability :: float},
            params :: %{(name :: String.t()) => value :: String.t()}}
   def predict_intent(model, context, x) do
     # tokenize the utterance
     [tokens] = Pipeline.transform(model.tokenizer, context, [x])
 
-    # predict the intent name
-    [intent] = Pipeline.predict_class(model.classifier, context, [tokens])
+    # predict the intent probabilities
+    [intents] =
+      Pipeline.predict_probability(model.classifier, context, [tokens])
 
     # predict the tag sequence
+    {intent, _probability} = Enum.max_by(intents, fn {_, v} -> v end)
```

Comment:
Would it make sense to have the NIF return an ordered list of tuples instead of a map to make this bit faster? I guess you'd never expect the cardinality of the class set to be anywhere near big enough for iteration to cause a slowdown; just thought I'd check about the complexity in making it a tiny bit faster.

Reply:
If elixir had a sorted map or array map, I would use it here, but the map data structure is more idiomatic of what it is actually returning. sklearn returns a list of probabilities and you have to constantly worry about which one corresponds to which class. If we have enough classes for this to be a performance problem, we will need more horsepower elsewhere anyway.

Reply:
Fair points all.